src/GitHub.Exports/Services/GitCloneCommandLineService.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Interop;
+using GitHub.Primitives;
+using Rothko;
+
+namespace GitHub.Services
+{
+    [Export(typeof(IGitCloneCommandLineService))]
+    public class GitCloneCommandLineService : IGitCloneCommandLineService
+    {
+        IVsAppCommandLine vsAppCommandLine;
+        IVSGitServices vsGitServices;
+        IVSServices vsServices;
+        IOperatingSystem operatingSystem;
+
+        [ImportingConstructor]
+        internal GitCloneCommandLineService(IGitHubServiceProvider sp, IVSGitServices vsGitServices,
+            ITeamExplorerServices teamExplorerServices, IVSServices vsServices, IOperatingSystem operatingSystem)
+            : this(sp.TryGetService<IVsAppCommandLine>(), vsGitServices, teamExplorerServices, vsServices, operatingSystem)
+        {
+        }
+
+        public GitCloneCommandLineService(IVsAppCommandLine vsAppCommandLine, IVSGitServices vsGitServices,
+            ITeamExplorerServices teamExplorerServices, IVSServices vsServices, IOperatingSystem operatingSystem)
+        {
+            this.vsAppCommandLine = vsAppCommandLine;
+            this.vsGitServices = vsGitServices;
+            this.vsServices = vsServices;
+            this.operatingSystem = operatingSystem;
+        }
+
+        public string FindGitCloneOption()
+        {
+
+            if (vsAppCommandLine == null)
+            {
+                return null;
+            }
+
+            int isPresent;
+            string optionValue;
+            if (ErrorHandler.Failed(vsAppCommandLine.GetOption("GitClone", out isPresent, out optionValue)))
```

Comment:
This "GitClone" string and the one in the `ProvideAppCommandLine` package atributes should probably be a const shared by both, so that it doesn't go out of sync. Not that it will, but given that it's likely we'll implement more of these and this code will likely be copy pasted around, it's good practice to avoid mistakes.